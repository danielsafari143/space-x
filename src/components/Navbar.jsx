import React from 'react';
import { NavLink } from 'react-router-dom';
import planet from '../assets/img/planet.png';

const Navbar = () => (
  <header>
    <div className="header">
      <h1 className="title">
        <img src={planet} alt="logo" />
        Space Travelers
      </h1>
      <nav>
        <NavLink
          to="/"
          className={({ isActive }) => (isActive ? 'active link' : 'pending link')}
        >
          Rockets
        </NavLink>
        <NavLink className="link" to="/missions">Missions</NavLink>
        <hr className="nav-hr" />
        <NavLink className="link" to="/myprofile">My Profile</NavLink>
      </nav>
    </div>
    <hr className="header-hr" />
  </header>
);

export default Navbar;
