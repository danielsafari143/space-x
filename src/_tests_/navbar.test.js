import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';
import Navbar from '../components/Navbar';

describe('should test the navbar', () => {
  it('should test the snapshot of the navbar', () => {
    const navb = renderer.create(<MemoryRouter><Navbar /></MemoryRouter>).toJSON();
    expect(navb).toMatchSnapshot();
  });
});
