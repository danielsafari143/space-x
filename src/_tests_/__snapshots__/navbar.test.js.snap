// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should test the navbar should test the snapshot of the navbar 1`] = `
<header>
  <div
    className="header"
  >
    <h1
      className="title"
    >
      Space Travelers
    </h1>
    <nav>
      <a
        aria-current="page"
        className="link active"
        href="/"
        onClick={[Function]}
      >
        Rockets
      </a>
      <a
        className="link"
        href="/missions"
        onClick={[Function]}
      >
        Missions
      </a>
      <hr
        className="nav-hr"
      />
      <a
        className="link"
        href="/myprofile"
        onClick={[Function]}
      >
        My Profile
      </a>
    </nav>
  </div>
  <hr
    className="header-hr"
  />
</header>
`;
